docs/_guide/properties.md

```diff
@@ -192,37 +316,102 @@ myProp: { attribute: 'my-prop' }
 
 ```js
-// No observed attribute for you
+// No observed attribute for this property
 myProp: { attribute: false }
 ```
 
-**Example: Configuring observed attributes**
+An observed attribute can be used to provide an initial value for a property via markup. See [Initialize properties with attributes in markup](#initialize-markup).
+
+**Example: Configure observed attributes**
+
+```js
+{% include projects/properties/attributeobserve/my-element.js %}
+```
+
+{% include project.html folder="properties/attributeobserve" openFile="my-element.js" %}
+
+### Configure reflected attributes {#reflected-attributes}
+
+You can configure a property so that whenever it changes, its value is reflected to its [observed attribute](#observed-attributes). For example:
+
+```js
+// Value of property "myProp" will reflect to attribute "myprop"
+myProp: { reflect: true }
+```
+
+When the property changes, LitElement uses the `toAttribute` function in the property's converter to set the attribute value from the new property value. 
+
+* If `toAttribute` returns `null`, the attribute is removed.
+
+* If `toAttribute` returns `undefined`, the attribute is not changed.
+
+* If `toAttribute` itself is undefined, the property value is set to the attribute value without conversion.
+
+{:.alert .alert-info}
+<div>
+
+**LitElement tracks reflection state during updates.** LitElement keeps track of  state information to avoid creating an infinite loop of updates between a property and an observed, reflected attribute.
+
+</div>
+
+**Example: Configure reflected attributes**
+
+```js
+{% include projects/properties/attributereflect/my-element.js %}
+```
+
+{% include project.html folder="properties/attributereflect" openFile="my-element.js" %}
+
+## Configure property accessors {#accessors}
+
+By default, LitElement generates a property accessor for all declared properties. The accessor is invoked whenever you set the property:
+
+```js
+// Declare a property
+static get properties() { return { myProp: { type: String } }; }
+...
+// Later, set the property
+this.myProp = 'hi'; // invokes myProp's generated property accessor
+```
+
+Generated accessors automatically call `requestUpdate`, initiating an update if one has not already begun.
+
+### Create custom property accessors {#accessors-custom}
+
+To create custom property accessors:
 
 ```js
-{% include projects/docs/properties/attribute/my-element.js %}
+static get properties() { return { myProp: { type: String } }; }
+set myProp(newValue) { 
+  ...
+  this.requestUpdate('foo', oldVal);
+}
+get myProp() { ... }
 ```
 
-### Configure reflection to attributes
+Call `requestUpdate('propertyName', oldValue)` when implementing a custom property setter to ensure that changes to the property will trigger updates when required. Pass the name and old value of the property to `requestUpdate` so that any property options can be correctly applied.
```

Comment:
So with the recent change, manually call `requestUpdate` is _only_ required in the `noAccessor` case when we don't generate an accessor for them (should be rare), so I think this can move down to that section.  It's obviously only required if they want the property to trigger an update.

Unless `noAccessor` is true, If a user writes an accessor with the same name as a declared property, a generated accessor is still created that calls `requestUpdate`, but it will "super" to the user accessor... similar to this:

```js
// The original user accessor is saved off (`userAccessor`, with `get` and
// `set` functions), and then the user accessor is overwritten with the generated
// accessor, which "supers" to the saved accessor; looks roughly like this:
set foo(value) {
  const oldValue = this.foo;
  this.requestUpdate('foo' oldValue);
  userAccessor.set.call(this, value);
}
get foo() {
  return userAccessor.get.call(this);
}
```
I think 👆is probably too much implementation detail to put in code, not sure.  But thought it would be useful to understand what we do under the covers to make sure it aligns with the docs.

Reply:
Thanks, that helps a lot!